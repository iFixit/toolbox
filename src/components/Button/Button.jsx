import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import constants from '../../constants';

const {
   borderRadius,
   color,
   fontSize,
   lineHeight,
   spacing,
   transition,
} = constants;

const propTypes = {
   /** Contents of the button. */
   children: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.node,
      PropTypes.arrayOf(PropTypes.node),
   ]).isRequired,
   /** Choose a button design. */
   design: PropTypes.oneOf([
      'default',
      'primary',
      'secondary',
      'outline',
      'plain',
   ]),
   /** Disable button. */
   disabled: PropTypes.bool,
   /** Set width of the button to 100%. */
   fullWidth: PropTypes.bool,
   /** If an `href` is provided, the underlying element will be an `<a>` instead of a `<button>`. */
   href: PropTypes.string,
   /** Size of button. */
   size: PropTypes.oneOf(['small', 'medium', 'large']),
   /** Callback when clicked. */
   onClick: PropTypes.func,
};

const defaultProps = {
   design: 'default',
   disabled: false,
   fullWidth: false,
   href: '',
   size: 'medium',
   onClick: () => {},
};

const designs = {
   default: {
      color: color.grayAlpha[9],
      backgroundColor: 'transparent',
      borderColor: color.grayAlpha[3],
      ':hover': {
         backgroundColor: color.grayAlpha[1],
      },
      ':focus': {
         boxShadow: `0 0 0 3px ${color.grayAlpha[3]}`,
      },
   },
   primary: {
      color: color.white,
      backgroundColor: color.blue[4],
      borderColor: color.blue[4],
      ':hover': {
         color: color.white,
         backgroundColor: color.blue[5],
         borderColor: color.blue[5],
      },
      ':focus': {
         boxShadow: `0 0 0 3px ${color.blueAlpha[2]}`,
      },
   },
   secondary: {
      color: color.grayAlpha[9],
      backgroundColor: color.grayAlpha[3],
      ':hover': {
         backgroundColor: color.grayAlpha[4],
      },
      ':focus': {
         boxShadow: `0 0 0 3px ${color.grayAlpha[4]}`,
      },
   },
   outline: {
      color: color.blue[4],
      backgroundColor: 'transparent',
      borderColor: color.blue[4],
      ':hover': {
         color: color.white,
         backgroundColor: color.blue[4],
         borderColor: color.blue[4],
      },
      ':focus': {
         boxShadow: `0 0 0 3px ${color.blueAlpha[2]}`,
      },
   },
   plain: {
      color: color.grayAlpha[9],
      backgroundColor: 'transparent',
      borderColor: 'transparent',
      ':hover': {
         color: color.blue[4],
      },
      ':focus': {
         backgroundColor: color.grayAlpha[1],
      },
   },
};

const sizes = {
   small: {
      padding: `${spacing[0]} ${spacing[2]}`,
      fontSize: fontSize[1],
      lineHeight: lineHeight.copy,
   },
   medium: {
      padding: `${spacing[1]} ${spacing[3]}`,
      fontSize: fontSize[1],
      lineHeight: lineHeight.copy,
   },
   large: {
      padding: `${spacing[3]} ${spacing[4]}`,
      fontSize: fontSize[2],
      lineHeight: lineHeight.title,
   },
};

const ButtonContainer = styled.button(
   {
      display: 'inline-block',
      boxSizing: 'border-box',
      fontFamily: 'inherit',
      textDecoration: 'none',
      border: '1px solid transparent',
      borderRadius,
      outline: 0,
      whiteSpace: 'nowrap',
      cursor: 'pointer',
      userSelect: 'none',
      transition: `all ${transition.duration} ${transition.easing}`,
   },
   ({ design }) => designs[design],
   ({ size }) => sizes[size],
   ({ fullWidth }) =>
      fullWidth && {
         display: 'flex',
         width: '100%',
      },
   ({ disabled }) =>
      disabled && {
         opacity: 0.5,
         pointerEvents: 'none',
      },
);

const LinkContainer = ButtonContainer.withComponent('a');

// Buttons cannot be flex containers in some browsers (i.e. Safari).
// To get around this, we wrap the contents of the button in a flex container.
const FlexContainer = styled.span({
   width: '100%',
   display: 'flex',
   justifyContent: 'center',
   alignItems: 'center',
   textAlign: 'center',

   '> :not(:first-child)': {
      marginLeft: spacing[1],
   },
});

const TruncateContainer = styled.span({
   textOverflow: 'ellipsis',
   overflow: 'hidden',
   whiteSpace: 'nowrap',
});

const Button = ({ children, ...props }) => {
   const Container = props.href === '' ? ButtonContainer : LinkContainer;

   return (
      <Container {...props} tabIndex={props.disabled ? -1 : 0}>
         <FlexContainer>
            {React.Children.map(
               children,
               child =>
                  typeof child !== 'string' ? (
                     child
                  ) : (
                     <TruncateContainer>{child}</TruncateContainer>
                  ),
            )}
         </FlexContainer>
      </Container>
   );
};

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
