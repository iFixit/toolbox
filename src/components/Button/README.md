### Simple Button

```
<Button>Button</Button>
```

### Button Designs

```
<Button>Default</Button>
```

```
<Button design="primary">Primary</Button>
```

```
<Button design="outline">Outline</Button>
```

### Button Sizes

```
<Button size="small">Small</Button>
```

```
<Button>Medium</Button>
```

```
<Button size="large">Large</Button>
```

### Full-Width Button

```
<Button fullWidth>Full-width</Button>
```

### Disabled Button

```
<Button disabled>Disabled</Button>
```

### onClick Callback

```
<Button onClick={() => console.log('clicked')}>Click me</Button>
```

### Link

```
<Button link href="https://ifixit.com" target="_blank">Go to iFixit.com</Button>
```
