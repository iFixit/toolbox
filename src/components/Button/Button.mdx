---
name: Button
route: /components/button
menu: Components
---

import Button from './Button'
import Icon from '../Icon/Icon'
import { Playground, PropsTable } from 'docz'

# Button

## Examples

### Default

<Playground>
  <Button>Button</Button>
</Playground>

### Button designs

<Playground>
  <Button design="default">Default</Button>
</Playground>

<Playground>
  <Button design="primary">Primary</Button>
</Playground>

<Playground>
  <Button design="secondary">Secondary</Button>
</Playground>

<Playground>
  <Button design="outline">Outline</Button>
</Playground>

<Playground>
  <Button design="plain">Plain</Button>
</Playground>

### Icon buttons

<Playground>
  <Button>
    <Icon name="heart" size={16} />
    <span>Icon left</span>
  </Button>
</Playground>

<Playground>
  <Button>
    <span>Icon right</span>
    <Icon name="search" size={16} />
  </Button>
</Playground>

### Button sizes

<Playground>
  <Button size="small">Small</Button>
</Playground>

<Playground>
  <Button size="medium">Medium</Button>
</Playground>

<Playground>
  <Button size="large">Large</Button>
</Playground>

### Full-width Button

<Playground>
  <Button fullWidth>Full-width</Button>
</Playground>

### Disabled Button

<Playground>
  <Button disabled>Disabled</Button>
</Playground>

### onClick callback

<Playground>
  <Button onClick={() => console.log('clicked')}>Click me</Button>
</Playground>

### Link

<Playground>
  <Button href="https://ifixit.com" target="_blank">Go to iFixit.com</Button>
</Playground>

## Props

<PropsTable of={Button} />
