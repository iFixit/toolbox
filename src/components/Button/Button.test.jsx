import React from 'react';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from './Button';

configure({ adapter: new Adapter() });

test('renders without crashing', () => {
   const tree = renderer.create(<Button>Test Button</Button>).toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders when disabled', () => {
   const tree = renderer
      .create(<Button disabled>Disabled Button</Button>)
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with default design', () => {
   const tree = renderer
      .create(<Button design="default">Default Button</Button>)
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with primary design', () => {
   const tree = renderer
      .create(<Button design="primary">Primary Button</Button>)
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with secondary design', () => {
   const tree = renderer
      .create(<Button design="secondary">Secondary Button</Button>)
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with outline design', () => {
   const tree = renderer
      .create(<Button design="outline">Outline Button</Button>)
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with plain design', () => {
   const tree = renderer
      .create(<Button design="plain">Plain Button</Button>)
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with small size', () => {
   const tree = renderer
      .create(<Button size="small">Small Button</Button>)
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with large size', () => {
   const tree = renderer
      .create(<Button size="large">Large Button</Button>)
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders at fullWidth', () => {
   const tree = renderer
      .create(<Button fullWidth>Full Width Button</Button>)
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with a when href is present', () => {
   const tree = renderer
      .create(
         <Button href="https://ifixit.com" target="_blank">
            Go to iFixit.com
         </Button>,
      )
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('calls onClick when button is clicked', () => {
   const handleClick = jest.fn();
   const wrapper = mount(
      <Button onClick={handleClick}>Clickable Button</Button>,
   );
   wrapper.find('button').simulate('click');
   expect(handleClick).toHaveBeenCalledTimes(1);
});
