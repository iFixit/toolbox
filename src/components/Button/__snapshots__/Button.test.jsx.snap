// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders at fullWidth 1`] = `
.glamor-1,
[data-glamor-1] {
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1> :not(:first-child),
[data-glamor-1]> :not(:first-child) {
  margin-left: 8px;
}

.glamor-0,
[data-glamor-0] {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.glamor-2,
[data-glamor-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  box-sizing: border-box;
  font-family: inherit;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: 0;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  transition: all 150ms ease-in-out;
  color: rgba(0, 3, 6, 0.87);
  background-color: transparent;
  border-color: rgba(0, 3, 6, 0.12);
  padding: 8px 16px;
  font-size: 14px;
  line-height: 1.5;
  width: 100%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover {
  background-color: rgba(0, 3, 6, 0.04);
}

.glamor-2:focus,
[data-glamor-2]:focus {
  box-shadow: 0 0 0 3px rgba(0, 3, 6, 0.12);
}

<button
  className="glamor-2"
  disabled={false}
  onClick={[Function]}
  tabIndex={0}
>
  <span
    className="glamor-1"
  >
    <span
      className="glamor-0"
    >
      Full Width Button
    </span>
  </span>
</button>
`;

exports[`renders when disabled 1`] = `
.glamor-1,
[data-glamor-1] {
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1> :not(:first-child),
[data-glamor-1]> :not(:first-child) {
  margin-left: 8px;
}

.glamor-0,
[data-glamor-0] {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.glamor-2,
[data-glamor-2] {
  display: inline-block;
  box-sizing: border-box;
  font-family: inherit;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: 0;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  transition: all 150ms ease-in-out;
  color: rgba(0, 3, 6, 0.87);
  background-color: transparent;
  border-color: rgba(0, 3, 6, 0.12);
  padding: 8px 16px;
  font-size: 14px;
  line-height: 1.5;
  opacity: 0.5;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover {
  background-color: rgba(0, 3, 6, 0.04);
}

.glamor-2:focus,
[data-glamor-2]:focus {
  box-shadow: 0 0 0 3px rgba(0, 3, 6, 0.12);
}

<button
  className="glamor-2"
  disabled={true}
  onClick={[Function]}
  tabIndex={-1}
>
  <span
    className="glamor-1"
  >
    <span
      className="glamor-0"
    >
      Disabled Button
    </span>
  </span>
</button>
`;

exports[`renders with a when href is present 1`] = `
.glamor-2,
[data-glamor-2] {
  display: inline-block;
  box-sizing: border-box;
  font-family: inherit;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: 0;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  transition: all 150ms ease-in-out;
  color: rgba(0, 3, 6, 0.87);
  background-color: transparent;
  border-color: rgba(0, 3, 6, 0.12);
  padding: 8px 16px;
  font-size: 14px;
  line-height: 1.5;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover {
  background-color: rgba(0, 3, 6, 0.04);
}

.glamor-2:focus,
[data-glamor-2]:focus {
  box-shadow: 0 0 0 3px rgba(0, 3, 6, 0.12);
}

.glamor-1,
[data-glamor-1] {
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1> :not(:first-child),
[data-glamor-1]> :not(:first-child) {
  margin-left: 8px;
}

.glamor-0,
[data-glamor-0] {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

<a
  className="glamor-2"
  href="https://ifixit.com"
  onClick={[Function]}
  tabIndex={0}
  target="_blank"
>
  <span
    className="glamor-1"
  >
    <span
      className="glamor-0"
    >
      Go to iFixit.com
    </span>
  </span>
</a>
`;

exports[`renders with default design 1`] = `
.glamor-2,
[data-glamor-2] {
  display: inline-block;
  box-sizing: border-box;
  font-family: inherit;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: 0;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  transition: all 150ms ease-in-out;
  color: rgba(0, 3, 6, 0.87);
  background-color: transparent;
  border-color: rgba(0, 3, 6, 0.12);
  padding: 8px 16px;
  font-size: 14px;
  line-height: 1.5;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover {
  background-color: rgba(0, 3, 6, 0.04);
}

.glamor-2:focus,
[data-glamor-2]:focus {
  box-shadow: 0 0 0 3px rgba(0, 3, 6, 0.12);
}

.glamor-1,
[data-glamor-1] {
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1> :not(:first-child),
[data-glamor-1]> :not(:first-child) {
  margin-left: 8px;
}

.glamor-0,
[data-glamor-0] {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

<button
  className="glamor-2"
  disabled={false}
  onClick={[Function]}
  tabIndex={0}
>
  <span
    className="glamor-1"
  >
    <span
      className="glamor-0"
    >
      Default Button
    </span>
  </span>
</button>
`;

exports[`renders with large size 1`] = `
.glamor-1,
[data-glamor-1] {
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1> :not(:first-child),
[data-glamor-1]> :not(:first-child) {
  margin-left: 8px;
}

.glamor-0,
[data-glamor-0] {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.glamor-2,
[data-glamor-2] {
  display: inline-block;
  box-sizing: border-box;
  font-family: inherit;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: 0;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  transition: all 150ms ease-in-out;
  color: rgba(0, 3, 6, 0.87);
  background-color: transparent;
  border-color: rgba(0, 3, 6, 0.12);
  padding: 16px 24px;
  font-size: 16px;
  line-height: 1.25;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover {
  background-color: rgba(0, 3, 6, 0.04);
}

.glamor-2:focus,
[data-glamor-2]:focus {
  box-shadow: 0 0 0 3px rgba(0, 3, 6, 0.12);
}

<button
  className="glamor-2"
  disabled={false}
  onClick={[Function]}
  tabIndex={0}
>
  <span
    className="glamor-1"
  >
    <span
      className="glamor-0"
    >
      Large Button
    </span>
  </span>
</button>
`;

exports[`renders with outline design 1`] = `
.glamor-1,
[data-glamor-1] {
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1> :not(:first-child),
[data-glamor-1]> :not(:first-child) {
  margin-left: 8px;
}

.glamor-0,
[data-glamor-0] {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.glamor-2,
[data-glamor-2] {
  display: inline-block;
  box-sizing: border-box;
  font-family: inherit;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: 0;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  transition: all 150ms ease-in-out;
  color: #0071CE;
  background-color: transparent;
  border-color: #0071CE;
  padding: 8px 16px;
  font-size: 14px;
  line-height: 1.5;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover {
  color: #FFFFFF;
  background-color: #0071CE;
  border-color: #0071CE;
}

.glamor-2:focus,
[data-glamor-2]:focus {
  box-shadow: 0 0 0 3px rgba(0, 113, 206, 0.32);
}

<button
  className="glamor-2"
  disabled={false}
  onClick={[Function]}
  tabIndex={0}
>
  <span
    className="glamor-1"
  >
    <span
      className="glamor-0"
    >
      Outline Button
    </span>
  </span>
</button>
`;

exports[`renders with plain design 1`] = `
.glamor-1,
[data-glamor-1] {
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1> :not(:first-child),
[data-glamor-1]> :not(:first-child) {
  margin-left: 8px;
}

.glamor-0,
[data-glamor-0] {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.glamor-2,
[data-glamor-2] {
  display: inline-block;
  box-sizing: border-box;
  font-family: inherit;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: 0;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  transition: all 150ms ease-in-out;
  color: rgba(0, 3, 6, 0.87);
  background-color: transparent;
  border-color: transparent;
  padding: 8px 16px;
  font-size: 14px;
  line-height: 1.5;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover {
  color: #0071CE;
}

.glamor-2:focus,
[data-glamor-2]:focus {
  background-color: rgba(0, 3, 6, 0.04);
}

<button
  className="glamor-2"
  disabled={false}
  onClick={[Function]}
  tabIndex={0}
>
  <span
    className="glamor-1"
  >
    <span
      className="glamor-0"
    >
      Plain Button
    </span>
  </span>
</button>
`;

exports[`renders with primary design 1`] = `
.glamor-1,
[data-glamor-1] {
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1> :not(:first-child),
[data-glamor-1]> :not(:first-child) {
  margin-left: 8px;
}

.glamor-0,
[data-glamor-0] {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.glamor-2,
[data-glamor-2] {
  display: inline-block;
  box-sizing: border-box;
  font-family: inherit;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: 0;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  transition: all 150ms ease-in-out;
  color: #FFFFFF;
  background-color: #0071CE;
  border-color: #0071CE;
  padding: 8px 16px;
  font-size: 14px;
  line-height: 1.5;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover {
  color: #FFFFFF;
  background-color: #005091;
  border-color: #005091;
}

.glamor-2:focus,
[data-glamor-2]:focus {
  box-shadow: 0 0 0 3px rgba(0, 113, 206, 0.32);
}

<button
  className="glamor-2"
  disabled={false}
  onClick={[Function]}
  tabIndex={0}
>
  <span
    className="glamor-1"
  >
    <span
      className="glamor-0"
    >
      Primary Button
    </span>
  </span>
</button>
`;

exports[`renders with secondary design 1`] = `
.glamor-1,
[data-glamor-1] {
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1> :not(:first-child),
[data-glamor-1]> :not(:first-child) {
  margin-left: 8px;
}

.glamor-0,
[data-glamor-0] {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.glamor-2,
[data-glamor-2] {
  display: inline-block;
  box-sizing: border-box;
  font-family: inherit;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: 0;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  transition: all 150ms ease-in-out;
  color: rgba(0, 3, 6, 0.87);
  background-color: rgba(0, 3, 6, 0.12);
  padding: 8px 16px;
  font-size: 14px;
  line-height: 1.5;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover {
  background-color: rgba(0, 3, 6, 0.26);
}

.glamor-2:focus,
[data-glamor-2]:focus {
  box-shadow: 0 0 0 3px rgba(0, 3, 6, 0.26);
}

<button
  className="glamor-2"
  disabled={false}
  onClick={[Function]}
  tabIndex={0}
>
  <span
    className="glamor-1"
  >
    <span
      className="glamor-0"
    >
      Secondary Button
    </span>
  </span>
</button>
`;

exports[`renders with small size 1`] = `
.glamor-1,
[data-glamor-1] {
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1> :not(:first-child),
[data-glamor-1]> :not(:first-child) {
  margin-left: 8px;
}

.glamor-0,
[data-glamor-0] {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.glamor-2,
[data-glamor-2] {
  display: inline-block;
  box-sizing: border-box;
  font-family: inherit;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: 0;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  transition: all 150ms ease-in-out;
  color: rgba(0, 3, 6, 0.87);
  background-color: transparent;
  border-color: rgba(0, 3, 6, 0.12);
  padding: 4px 12px;
  font-size: 14px;
  line-height: 1.5;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover {
  background-color: rgba(0, 3, 6, 0.04);
}

.glamor-2:focus,
[data-glamor-2]:focus {
  box-shadow: 0 0 0 3px rgba(0, 3, 6, 0.12);
}

<button
  className="glamor-2"
  disabled={false}
  onClick={[Function]}
  tabIndex={0}
>
  <span
    className="glamor-1"
  >
    <span
      className="glamor-0"
    >
      Small Button
    </span>
  </span>
</button>
`;

exports[`renders without crashing 1`] = `
.glamor-2,
[data-glamor-2] {
  display: inline-block;
  box-sizing: border-box;
  font-family: inherit;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: 0;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  transition: all 150ms ease-in-out;
  color: rgba(0, 3, 6, 0.87);
  background-color: transparent;
  border-color: rgba(0, 3, 6, 0.12);
  padding: 8px 16px;
  font-size: 14px;
  line-height: 1.5;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover {
  background-color: rgba(0, 3, 6, 0.04);
}

.glamor-2:focus,
[data-glamor-2]:focus {
  box-shadow: 0 0 0 3px rgba(0, 3, 6, 0.12);
}

.glamor-1,
[data-glamor-1] {
  width: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1> :not(:first-child),
[data-glamor-1]> :not(:first-child) {
  margin-left: 8px;
}

.glamor-0,
[data-glamor-0] {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

<button
  className="glamor-2"
  disabled={false}
  onClick={[Function]}
  tabIndex={0}
>
  <span
    className="glamor-1"
  >
    <span
      className="glamor-0"
    >
      Test Button
    </span>
  </span>
</button>
`;
